pkg/util/wait_group_wrapper.go

```diff
@@ -56,7 +59,7 @@ func (w *WaitGroupEnhancedWrapper) checkUnExitedProcess(exit chan struct{}) {
 	<-exit
 	logutil.BgLogger().Info("waitGroupWrapper start exit-checking", zap.String("source", w.source))
 	if w.check() {
-		ticker := time.NewTimer(2 * time.Second)
+		ticker := time.NewTicker(2 * time.Second)
```

Comment:
This is a bug fix, it should be a ticker instead of a timer.
Before the fix, the log only print twice ..... 

After fix, it works and helps me to find the unexited services:

```
[2024/12/25 15:09:52.072 +08:00] [WARN] [wait_group_wrapper.go:78] ["background process unexited while received exited signal"] [process="[PlanReplayerTaskDumpHandle,analyzeJobsCleanupWorker,PlanReplayerTaskCollectHandle,autoAnalyzeWorker,loadStatsWorker,asyncLoadHistogram,runawayWatchSyncLoop,mdlCheckLoop,logBackupAdvancer,LoadSysVarCacheLoop,globalConfigSyncerKeeper,closestReplicaReadCheckLoop,updateStatsWorker,loadSigningCertLoop,HistoricalStatsWorker,indexUsageWorker,topNSlowQueryLoop,infoSyncerKeeper,requestUnitsWriterLoop,topologySyncerKeeper,loadPrivilegeInLoop,loadSchemaInLoop,globalBindHandleWorkerLoop,distTaskFrameworkLoop]"] [source=domain]
[2024/12/25 15:09:54.073 +08:00] [WARN] [wait_group_wrapper.go:78] ["background process unexited while received exited signal"] [process="[loadSchemaInLoop,globalBindHandleWorkerLoop,PlanReplayerTaskCollectHandle,autoAnalyzeWorker,runawayWatchSyncLoop,updateStatsWorker]"] [source=domain]
[2024/12/25 16:17:02.259 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[loadSchemaInLoop,distTaskFrameworkLoop,logBackupAdvancer,PlanReplayerTaskDumpHandle,planCacheEvictTrigger,globalConfigSyncerKeeper,closestReplicaReadCheckLoop,LoadSysVarCacheLoop,globalBindHandleWorkerLoop,analyzeJobsCleanupWorker,PlanReplayerTaskCollectHandle,updateStatsWorker,topNSlowQueryLoop,loadPrivilegeInLoop,runawayWatchSyncLoop,HistoricalStatsWorker,dumpFileGcChecker,loadStatsWorker,autoAnalyzeWorker]"] [source=domain]
[2024/12/25 16:17:04.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[LoadSysVarCacheLoop,globalBindHandleWorkerLoop,analyzeJobsCleanupWorker,PlanReplayerTaskCollectHandle,updateStatsWorker,runawayWatchSyncLoop,loadStatsWorker,autoAnalyzeWorker,loadSchemaInLoop]"] [source=domain]
[2024/12/25 16:17:06.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[globalBindHandleWorkerLoop,analyzeJobsCleanupWorker,PlanReplayerTaskCollectHandle,updateStatsWorker,runawayWatchSyncLoop,loadStatsWorker,loadSchemaInLoop,LoadSysVarCacheLoop]"] [source=domain]
[2024/12/25 16:17:08.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[loadSchemaInLoop,LoadSysVarCacheLoop,globalBindHandleWorkerLoop,analyzeJobsCleanupWorker,PlanReplayerTaskCollectHandle,updateStatsWorker,runawayWatchSyncLoop,loadStatsWorker]"] [source=domain]
[2024/12/25 16:17:10.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[loadSchemaInLoop,LoadSysVarCacheLoop,globalBindHandleWorkerLoop,analyzeJobsCleanupWorker,PlanReplayerTaskCollectHandle,updateStatsWorker,runawayWatchSyncLoop,loadStatsWorker]"] [source=domain]
[2024/12/25 16:17:12.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[loadStatsWorker,loadSchemaInLoop,LoadSysVarCacheLoop,globalBindHandleWorkerLoop,analyzeJobsCleanupWorker,PlanReplayerTaskCollectHandle,updateStatsWorker,runawayWatchSyncLoop]"] [source=domain]
[2024/12/25 16:17:14.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[PlanReplayerTaskCollectHandle,updateStatsWorker,runawayWatchSyncLoop,loadStatsWorker,loadSchemaInLoop,LoadSysVarCacheLoop,globalBindHandleWorkerLoop,analyzeJobsCleanupWorker]"] [source=domain]
[2024/12/25 16:17:16.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[LoadSysVarCacheLoop,globalBindHandleWorkerLoop,PlanReplayerTaskCollectHandle,updateStatsWorker,runawayWatchSyncLoop,loadStatsWorker,loadSchemaInLoop]"] [source=domain]
[2024/12/25 16:17:18.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[loadStatsWorker,loadSchemaInLoop,LoadSysVarCacheLoop,globalBindHandleWorkerLoop,PlanReplayerTaskCollectHandle,updateStatsWorker,runawayWatchSyncLoop]"] [source=domain]
[2024/12/25 16:17:20.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[globalBindHandleWorkerLoop,PlanReplayerTaskCollectHandle,updateStatsWorker,runawayWatchSyncLoop,loadStatsWorker,loadSchemaInLoop,LoadSysVarCacheLoop]"] [source=domain]
[2024/12/25 16:17:22.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[PlanReplayerTaskCollectHandle,updateStatsWorker,runawayWatchSyncLoop,loadStatsWorker,loadSchemaInLoop,LoadSysVarCacheLoop,globalBindHandleWorkerLoop]"] [source=domain]
[2024/12/25 16:17:24.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[LoadSysVarCacheLoop,globalBindHandleWorkerLoop,PlanReplayerTaskCollectHandle,updateStatsWorker,runawayWatchSyncLoop,loadStatsWorker,loadSchemaInLoop]"] [source=domain]
[2024/12/25 16:17:26.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[loadSchemaInLoop,LoadSysVarCacheLoop,globalBindHandleWorkerLoop,PlanReplayerTaskCollectHandle,updateStatsWorker,runawayWatchSyncLoop,loadStatsWorker]"] [source=domain]
[2024/12/25 16:17:28.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[LoadSysVarCacheLoop,globalBindHandleWorkerLoop,PlanReplayerTaskCollectHandle,updateStatsWorker,runawayWatchSyncLoop,loadStatsWorker,loadSchemaInLoop]"] [source=domain]
[2024/12/25 16:17:30.261 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[runawayWatchSyncLoop,loadStatsWorker,loadSchemaInLoop,LoadSysVarCacheLoop,globalBindHandleWorkerLoop,PlanReplayerTaskCollectHandle,updateStatsWorker]"] [source=domain]
[2024/12/25 16:17:32.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[runawayWatchSyncLoop,loadStatsWorker,loadSchemaInLoop,globalBindHandleWorkerLoop,PlanReplayerTaskCollectHandle,updateStatsWorker]"] [source=domain]
[2024/12/25 16:17:34.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[loadStatsWorker,loadSchemaInLoop,globalBindHandleWorkerLoop,PlanReplayerTaskCollectHandle,updateStatsWorker,runawayWatchSyncLoop]"] [source=domain]
[2024/12/25 16:17:36.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[PlanReplayerTaskCollectHandle,updateStatsWorker,runawayWatchSyncLoop,loadStatsWorker,loadSchemaInLoop,globalBindHandleWorkerLoop]"] [source=domain]
[2024/12/25 16:17:38.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[PlanReplayerTaskCollectHandle,updateStatsWorker,runawayWatchSyncLoop,loadStatsWorker,loadSchemaInLoop,globalBindHandleWorkerLoop]"] [source=domain]
[2024/12/25 16:17:40.261 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[globalBindHandleWorkerLoop,PlanReplayerTaskCollectHandle,runawayWatchSyncLoop,loadStatsWorker,loadSchemaInLoop]"] [source=domain]
[2024/12/25 16:17:42.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[PlanReplayerTaskCollectHandle,runawayWatchSyncLoop,loadStatsWorker,loadSchemaInLoop,globalBindHandleWorkerLoop]"] [source=domain]
[2024/12/25 16:17:44.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[runawayWatchSyncLoop,loadSchemaInLoop,PlanReplayerTaskCollectHandle]"] [source=domain]
[2024/12/25 16:17:46.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[runawayWatchSyncLoop,loadSchemaInLoop,PlanReplayerTaskCollectHandle]"] [source=domain]
[2024/12/25 16:17:48.261 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[PlanReplayerTaskCollectHandle,runawayWatchSyncLoop,loadSchemaInLoop]"] [source=domain]
[2024/12/25 16:17:50.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[loadSchemaInLoop,PlanReplayerTaskCollectHandle,runawayWatchSyncLoop]"] [source=domain]
[2024/12/25 16:17:52.261 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[runawayWatchSyncLoop,loadSchemaInLoop]"] [source=domain]
[2024/12/25 16:17:54.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[runawayWatchSyncLoop,loadSchemaInLoop]"] [source=domain]
[2024/12/25 16:17:56.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[loadSchemaInLoop,runawayWatchSyncLoop]"] [source=domain]
[2024/12/25 16:17:58.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[loadSchemaInLoop,runawayWatchSyncLoop]"] [source=domain]
[2024/12/25 16:18:00.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[runawayWatchSyncLoop,loadSchemaInLoop]"] [source=domain]
[2024/12/25 16:18:02.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[runawayWatchSyncLoop,loadSchemaInLoop]"] [source=domain]
[2024/12/25 16:18:04.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[loadSchemaInLoop,runawayWatchSyncLoop]"] [source=domain]
[2024/12/25 16:18:06.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[loadSchemaInLoop,runawayWatchSyncLoop]"] [source=domain]
[2024/12/25 16:18:08.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process="[runawayWatchSyncLoop,loadSchemaInLoop]"] [source=domain]
[2024/12/25 16:18:10.260 +08:00] [WARN] [wait_group_wrapper.go:82] ["background process unexited while received exited signal"] [process
```